test/test.js

```diff
@@ -0,0 +1,99 @@
+import { trips, travelers, traveler, destinations } from '/js/mockData.js'
+import { getUserTrips, getUserExpenditures } from '../js'
+import { expect } from 'chai';
+
+before(function () {
+  trips, travelers, traveler, destinations
+})
+
+describe('Get User Trips', function () {
+  it('should return all relevant trips based on user ID', () => {
+    const { approvedTrips } = getUserTrips(4, trips)
+    expect(approvedTrips).to.deep.equal([
+      { date: "2022/09/16", destinationID: 2, duration: 8, id: 1, status: "approved", suggestedActivities: [], travelers: 1, userID: 4 }
+    ])
+  })
+
+  it('should handle an invalid userID', () => {
+    const noTrips = getUserTrips('!', trips)
+    expect(noTrips).to.equal('You must enter a valid ID, please.')
+  })
+
+  it('should handle no ID', () => {
+    const badID = getUserTrips('', trips)
+    expect(badID).to.equal('You must enter a valid ID, please.')
+  })
+
+  it('should handle a return without any trips', () => {
+    const noTrips = getUserTrips(10, trips)
+    expect(noTrips).to.equal('You have no trips on the itinerary.')
+  })
+
+  it('should handle a return of just one trip', () => {
+    const { approvedTrips } = getUserTrips(30, trips);
+    expect(approvedTrips).to.deep.equal([{ date: "2023/10/12", destinationID: 27, duration: 4, id: 6, status: "approved", suggestedActivities: ["museum", "theater"], travelers: 2, userID: 30 }])
+  })
+
+  it('should handle pending trips', () => {
+    const { pendingTrips } = getUserTrips(4, trips)
+    const pendingTrip = [
+      { date: "2023/08/05", destinationID: 8, duration: 6, id: 5, status: "pending", suggestedActivities: ["diving"], travelers: 1, userID: 4 }
+    ]
+    expect(pendingTrips).to.deep.equal(pendingTrip)
+  })
+
+  describe('Get User Expenditures', function () {
+    it('should return total money spent on all approved trips in the target year, plus 10%, based on UserID', () => {
+      const totalBill = getUserExpenditures(4, trips, destinations, 2023)
```

Comment:
This test function with getUserExpenditures has 4 params, but I'm only seeing 3 params in the same function in index.js